Feature/individual tweaks

LindsayYoung opened the pull request:
This is a methodology change that I proposed but wanted to run by @PaulClark2 and @jwchumley 

We are seeing a lot of entries that are coded as individuals when they have committee id numbers and are clearly not individuals. 

To reduce the errors, this checks if there is a committee id listed for the contributor. If there is a committee id and it is not the filing committee's id, we are not counting that record as an individual.

If this sounds good, I can update the documentation.

closes #1779

jwchumley commented:
When you say "Coded as Individuals" are you talking about the entity type or transaction code 15? In any case I think you can pretty reliably use other id != committee id as an indicator of a transaction from or to another committee on schedules A and B.

________________________________
From: Lindsay Young <notifications@github.com>
Sent: Wednesday, August 3, 2016 7:06:43 PM
To: 18F/openFEC
Cc: Jeff Chumley; Mention
Subject: [18F/openFEC] Feature/individual tweaks (#1866)


This is a methodology change that I proposed but wanted to run by @PaulClark2<https://github.com/PaulClark2> and @jwchumley<https://github.com/jwchumley>

We are seeing a lot of entries that are coded as individuals when they have committee id numbers and are clearly not individuals.

To reduce the errors, this checks if there is a committee id listed for the contributor. If there is a committee id and it is not the filing committee's id, we are not counting that record as an individual.

If this sounds good, I can update the documentation.

closes #1779<https://github.com/18F/openFEC/issues/1779>

________________________________
You can view, comment on, or merge this pull request online at:

  https://github.com/18F/openFEC/pull/1866

Commit Summary

  *   Takes unitemized totals out of state break downs fix #1561
  *   Merge branch 'develop' into feature/individual-tweaks
  *   First pass at weeding out committees, still need to account for committees that are individuals
  *   Merge branch 'develop' into feature/individual-tweaks
  *   Merge branch 'develop' into feature/individual-tweaks
  *   Adding a check for committee id in individuals, also updating tests.

File Changes

  *   M data/functions/individual.sql<https://github.com/18F/openFEC/pull/1866/files#diff-0> (32)
  *   M data/sql_incremental_aggregates/prepare_schedule_a_aggregate_employer.sql<https://github.com/18F/openFEC/pull/1866/files#diff-1> (4)
  *   M data/sql_incremental_aggregates/prepare_schedule_a_aggregate_occupation.sql<https://github.com/18F/openFEC/pull/1866/files#diff-2> (4)
  *   M data/sql_incremental_aggregates/prepare_schedule_a_aggregate_size.sql<https://github.com/18F/openFEC/pull/1866/files#diff-3> (4)
  *   M data/sql_incremental_aggregates/prepare_schedule_a_aggregate_state.sql<https://github.com/18F/openFEC/pull/1866/files#diff-4> (8)
  *   M data/sql_incremental_aggregates/prepare_schedule_a_aggregate_zip.sql<https://github.com/18F/openFEC/pull/1866/files#diff-5> (4)
  *   M tests/test_itemized.py<https://github.com/18F/openFEC/pull/1866/files#diff-6> (2)

Patch Links:

  *   https://github.com/18F/openFEC/pull/1866.patch
  *   https://github.com/18F/openFEC/pull/1866.diff

-
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub<https://github.com/18F/openFEC/pull/1866>, or mute the thread<https://github.com/notifications/unsubscribe-auth/AMCA7wMJEhTXINO5ByihoOuqsFFaPYovks5qcR8DgaJpZM4JcL5E>.

PaulClark2 commented:
`To reduce the errors, this checks if there is a committee id listed for the contributor. If there is a committee id and it is not the filing committee's id, we are not counting that record as an individual.`

earmarked (15E) and JF (15J) receipts and a few others will almost always have committee IDs in the other_id field and they are individual contributions.

LindsayYoung commented:
Thanks @PaulClark2 and @jwchumley 

- I have updated  the coded to include the additional types listed in #1837 
- I added an exception for line numbers that are expected to have committee id. Are there any other line numbers for individuals that are expected to include committee id.

here is the [updated function](https://github.com/18F/openFEC/blob/916834210cc3dd88539baf3f76bd77b97649a690/data/functions/individual.sql)

PaulClark2 commented:
Form 3X Line 17 can include individual contributions that might have an other_id.  I think your fuction already included Line 17.  These would be coded transactions -- the codes are included in https://github.com/18F/openFEC/pull/1837

LindsayYoung commented:
Just got the OK from Paul and Jeff, we will look at the data it produces more closely when it is applied to dev.

ccostino commented on data/functions/individual.sql:
```diff
@@ -34,3 +45,28 @@ begin
+
+
+-- unitemized contributions should not be included in the state breakdowns
+create or replace function is_unitemized(memo_text text) returns bool as $$
```
It looks like this was included in the other PR and is now resulting in merge conflicts here.

ccostino commented on data/sql_incremental_aggregates/prepare_schedule_a_aggregate_state.sql:
```diff
@@ -9,9 +9,11 @@ select
-where rpt_yr >= :START_YEAR_AGGREGATE
+where
+    rpt_yr >= :START_YEAR_AGGREGATE and
+    is_unitemized(memo_text::text) = false
```
Same here, I think this was also a part of the other PR already.

ccostino commented:
Thanks, @LindsayYoung!  I've addressed the merge conflicts, once the tests all pass in CI I'll merge and run the `update_functions`, `rebuild_aggregates`, and `update_schemas` tasks against `dev`.
